Adding doctests for parameter sweep tool using RO w/ energy recovery flowsheet (#128)
nawi-hub/proteuslib: docs/how_to_guides/how_to_use_parameter_sweep.rst

Once this sequence of setup steps is performed, the parameters to be varied should be identified with a dictionary:

.. testcode::

    sweep_params = dict()
    sweep_params['Feed Concentration'] = (m.fs.feed.flow_mass_phase_comp[0, 'Liq', 'NaCl'], 0.005, 0.155, 4)
    sweep_params['Water Recovery'] = (m.fs.product_recovery, 0.3, 0.7, 4)

where the basic pattern is ``dict_name['Short/Pretty-print Name'] = (m.path.to.model.variable, lower_limit, upper_limit, num_samples)``.
For example, "Feed Concentration", which is accessed in the model variable ``m.fs.feed.flow_mass_phase_comp[0, 'Liq', 'NaCl']``, is to be varied between 0.005 and 0.155 with 4 equally-spaced values, i.e., ``[0.005, 0.055, 0.105, 0.155]``.
In this case, the 2 parameters will each be varied across 4 values for a total of 16 combinations.
Note that there is no limit on the number of sweep variables specified or their resolution besides the practical limit of how long it will take to optimize using each combination of parameters (e.g., if 5 different variables are provided and each one is individually represented with 20 discrete values, the total number of combinations is 20^5 = 3.2 million!).

> After specifying the input parameters, the user should then specify output values on the flowsheet which will be reported in the summary CSV file, which has a similar format to the sweep parameters.

Reviewer: ```suggestion
After specifying the input parameters, the user should then specify output values on the flowsheet that will be reported in the summary CSV file, which has a similar format to the sweep parameters.
```

> For this RO flowsheet we'll report the levelized cost of water, the optimized RO area, and the associated pump pressures:

Author: ```suggestion
For this RO flowsheet we'll report the levelized cost of water, the optimized RO area, and the output pressure of pump 1:
```

.. testcode::

    outputs = dict()
    outputs['RO membrane area'] = m.fs.RO.area
    outputs['Pump 1 pressure'] = m.fs.P1.control_volume.properties_out[0].pressure
    outputs['Levelized Cost of Water'] = m.fs.costing.LCOW 

> Once the problem is setup and the parameters are identified, the parameter_sweep function can finally be invoked which will perform the adjustment and optimization of the model invoking each combination of variables specified above, saving to `results.csv`; utilizing the solve method defined in our flowsheet module.

Reviewer: ```suggestion
Once the problem is setup and the parameters are identified, the parameter_sweep function can finally be invoked which will perform the adjustment and optimization of the model using each combination of variables specified above and saving to `results.csv` (utilizing the solve method defined in our flowsheet module).
```
